src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -319,9 +319,13 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
                         $params = array_merge($params, $data[$paramName]);
                     }
                 } elseif ($allowed && !$ignored && (isset($data[$key]) || array_key_exists($key, $data))) {
-                    $params[] = $data[$key];
-                    // don't run set for a parameter passed to the constructor
-                    unset($data[$key]);
+                    $parameterData = $data[$key];
+                    if (null !== $constructogrParameter->getClass()) {
+                        $parameterData = $this->serializer->denormalize($parameterData, $constructorParameter->getClass()->getName(), null, $context);
```

Comment:
I would add `$format` as a new optional paremeter for `instantiateObject` and pass it to the serializer.